File: frontend/src/types/document.ts
/** API_CONTRACT.md §3 Documents */

export type DocumentType = 'cv' | 'linkedin_pdf' | 'generated_cv';

export type DocumentStatus = 'queued' | 'processing' | 'completed' | 'failed';

export interface ParsedSkills {
  languages: string[];
  frameworks: string[];
  tools: string[];
  experience: unknown[];
  education: unknown[];
}

export interface UploadDocumentResponse {
  task_id: string;
  document_id: string;
  document_type: DocumentType;
  status: DocumentStatus;
}

export interface DocumentStatusResponse {
  document_id: string;
  document_type: DocumentType;
  status: DocumentStatus;
  error_message: string | null;
}

export interface DocumentSummary {
  document_id: string;
  document_type: DocumentType;
  cv_score: number | null;
  cv_score_explanation: string | null;
  parsed_skills: ParsedSkills;
  status: DocumentStatus;
  uploaded_at: string;
}

export interface LatestDocumentsResponse {
  documents: DocumentSummary[];
}

export interface DocumentListResponse {
  items: DocumentSummary[];
  total: number;
}

File: backend/app/schemas/document.py
"""API_CONTRACT.md §3 Documents şemaları."""
from datetime import datetime

from pydantic import BaseModel


class ParsedSkills(BaseModel):
    languages: list[str] = []
    frameworks: list[str] = []
    tools: list[str] = []
    experience: list = []
    education: list = []


class UploadDocumentResponse(BaseModel):
    task_id: str
    document_id: str
    document_type: str
    status: str


class DocumentStatusResponse(BaseModel):
    document_id: str
    document_type: str
    status: str
    error_message: str | None = None


class DocumentSummary(BaseModel):
    document_id: str
    document_type: str
    cv_score: int | None
    cv_score_explanation: str | None
    parsed_skills: ParsedSkills
    status: str
    uploaded_at: datetime


class LatestDocumentsResponse(BaseModel):
    documents: list[DocumentSummary]


class DocumentListResponse(BaseModel):
    items: list[DocumentSummary]
    total: int
